tests/test_github_start_date.py

```diff
@@ -126,11 +126,11 @@ def run_test(self, date_1, date_2, streams):
                 # collect information for assertions from syncs 1 & 2 base on expected values
                 record_count_sync_1 = record_count_by_stream_1.get(stream, 0)
                 record_count_sync_2 = record_count_by_stream_2.get(stream, 0)
-                primary_keys_list_1 = [tuple(message.get('data').get(expected_pk) for expected_pk in expected_primary_keys)
-                                       for message in synced_records_1.get(stream).get('messages')
+                primary_keys_list_1 = [tuple(message.get('data', {}).get(expected_pk) for expected_pk in expected_primary_keys)
```

Comment:
Similar to the automatic fields, could you confirm this results in a list of tuple objects. If it yields only the generators I believe the test would still pass, but would not be testing the primary key values.

Reply:
confirmed in debugger, using` tuple()` explicitly instead of `()` should return a tuple